src/boot/efi/fdt.c

```diff
@@ -0,0 +1,1289 @@
+// SPDX-License-Identifier: (GPL-2.0-or-later OR BSD-2-Clause)
+/*
+ * Adapted from libfdt - Flat Device Tree manipulation
+ * Copyright (C) 2006 David Gibson, IBM Corporation.
+ * Copyright 2021 Emil Renner Berthing
+ */
+
+#include <stdbool.h>
+#include <efi.h>
+#include <efilib.h>
+
+#include "fdt.h"
+
+static inline void memcpy(void *restrict dest, const void *restrict src, size_t n)
+{
+        CopyMem(dest, src, n);
+}
```

Comment:
this whole file is totally not in-line with our coding style.

What's the story here? this sounds like basically unmaintanable to us. This is copied from elsewhere? Where from? If this is copied once, this is going to bitrot. Isn't there a better way to pull this in? git module? static linking? i mean, meson has nice submodule support, can't we use this?

Reply:
I know, please read the description above.